tutorials/accelerated-python/notebooks/fundamentals/03__numpy_to_cupy__ndarray_basics.ipynb

```diff
@@ -1,385 +1,509 @@
+      "execution_count": null,
+      "id": "de5bdefb",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "# CPU Operation\n",
+        "x_cpu *= 5"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "6a7f32b8",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "# GPU Operation\n",
+        "x_gpu *= 5\n",
+        "\n",
+        "cp.cuda.Stream.null.synchronize()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "bc24579f",
+      "metadata": {},
+      "source": [
+        "The GPU completes this operation notably faster, with the code staying the same."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "83c69334",
+      "metadata": {},
+      "source": [
+        "### Sequential Operations & Memory\n",
+        "\n",
+        "Now let's do a couple of operations sequentially, something which would suffer from memory transfer times in Numba examples without explicit memory management."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "c0294dbc",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "# CPU: Sequential math\n",
+        "x_cpu *= 5\n",
+        "x_cpu *= x_cpu\n",
+        "x_cpu += x_cpu"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "acafdbe7",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "# GPU: Sequential math\n",
+        "x_gpu *= 5\n",
+        "x_gpu *= x_gpu\n",
+        "x_gpu += x_gpu\n",
+        "\n",
+        "cp.cuda.Stream.null.synchronize()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "0f250bbb",
+      "metadata": {},
+      "source": [
+        "The GPU ran that much faster even without us explicitly managing memory. This is because CuPy is handling all of this for us transparently."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "84221268",
+      "metadata": {},
+      "source": [
+        "## 3. Complex Operations (Linear Algebra)\n",
+        "\n",
+        "GPUs excel at Linear Algebra. Let's look at **Singular Value Decomposition (SVD)**, a computationally heavy $O(N^3)$ operation."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "978af795",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "# CPU SVD\n",
+        "x_cpu = np.random.random((1000, 1000))\n",
+        "u, s, v = np.linalg.svd(x_cpu)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "e0bc855b",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "# GPU SVD\n",
+        "x_gpu = cp.random.random((1000, 1000))\n",
+        "u, s, v = cp.linalg.svd(x_gpu)"
```

Comment:
This is missing a stream synchronize. I think it may be best to use cupyx benchmark instead of %%time.